lib/tasks/check_fixity.rake

```diff
@@ -0,0 +1,35 @@
+namespace :scihist_digicoll do
+  desc """
+  Checks the fixity of all Assets in the database.
+  bundle exec rake scihist_digicoll:check_fixity
+  """
+  task :check_fixity => :environment do
+    asset_count = Asset.all.count
+    if asset_count == 0
+      abort ("No assets found to check.")
+    end
+    progress_bar = ProgressBar.create(total: asset_count, format: "%a %t: |%B| %R/s %c/%u %p%% %e")
+    progress_bar.log("INFO: checking asset fixity")
+    Asset.all.each do |asset|
+      FixityChecker.new(asset).check
+      progress_bar.increment
+    end
+  end
+
+  """
+  bundle exec rake scihist_digicoll:prune_fixity_checks
+  """
+  task :prune_fixity_checks => :environment do
+    asset_count = Asset.all.count
+    if asset_count == 0
+      abort ("No assets found to check.")
+    end
+    progress_bar = ProgressBar.create(total: asset_count, format: "%a %t: |%B| %R/s %c/%u %p%% %e")
+    progress_bar.log("INFO: pruning the logs of asset fixity checks")
+    Asset.all.each do |asset|
+      FixityChecker.new(asset).prune_checks
+      progress_bar.increment
+    end
+  end
```

Comment:
I don't think there should be a separate prune_fixity_checks task. Rather, in `check_fixity`, you should just prune after checking. Maybe this works?

```ruby
checker = FixityChecker.new(asset)
checker.check
checker.prune_checks
```

Is there any reason you wouldn't want to prune checks after checking in the rake task?

I think in chf_sufia, it even automatically pruned as _part_ of the `check` method, you got a prune every time you called check. Which I think a case could be made for -- is there any case at all we'd want to check without pruning? Say, we have a "check this asset now" button in admin UI -- you'd want to prune after adding a new check, and making it a separate method makes it easy to forget to do that. So maybe it should just be automatically called by `check` method -- but i'm not sure, it's okay to leave it separate for now, but I am feeling confident the `check_fixity` rake task should call it for each thing it checks. 

(Ah, I see how this interacts with the "sifting" stuff maybe? Yeah, let's talk about sifting stuff).

Reply:
Or maybe add a #check_and_prune method that just calls both? And the rake task (And most other places) can just call it. Header docs can suggest that's the one you want.

Reply:
```checker = FixityChecker.new(asset)
checker.check
checker.prune_checks``` is perfect, with some memoization.